### ISSUE TITLE
new type signatures

### ISSUE THREAD

**Issue by gittheking** [jsiebern/bs-material-ui#5]
Hey @jsiebern,

Great work as always, I'm really enjoying the new improvements in the library. I've recently upgraded from from `0.1.6` to `0.2.0` and was able to migrate everything rather seamlessly aside from a few things that are puzzling me. I'm currently building out some cascading menus so I'm making use of the properties `transformOrigin` and `anchorOrigin`, previously I was able to set values like:
```
<MaterialUi.Menu
  transformOrigin={horizontal: int, vertical: int}
  anchorOrigin={horizontal: int, vertical: int}
  ...
/>
```
however with the new type signatures, these values no longer work - it looks like you're moving towards using the enums/variant values which I very much agree with, but even attempting to pass record with something like: `{horizontal: `Left, vertical: `Center}` I'll still get type errors telling me they are expecting `MaterialUi.Menu.typeAnchorOrigin`. Looking through the source hasn't quite helped as much as it has in the past. Am I missing something here?

Thanks for all your hard work on this project!

**Comment by jsiebern** [comment]
Hi, thank you very much for your kind words! I have changed all objects to have a create function (at least if they contain optional values!). In the future I guess I'll better switch to abstract object types (BS 3.0.0) to make that more clear.

In the meantime, you can use the provided `makeTransformOrigin` like so:

```reason
<MaterialUi.Menu
  transformOrigin=(MaterialUi.Menu.makeTransformOrigin(~horizontal=30, ~vertical=30, ()))
  anchorOrigin=(MaterialUi.Menu.makeAnchorOrigin(~horizontal=30, ~vertical=30, ()))
/>
```

I didn't test it just now but something like this should work!

Do you think, as a user, that I should switch to [Abstract obj types](https://bucklescript.github.io/docs/en/generate-converters-accessors.html#abstract-type)?

**Comment by jsiebern** [comment]
(Looking at the generated code myself just now, I realized that the make function isn't properly typed for some reason, I'll look into that!)

**Comment by gittheking** [comment]
@jsiebern thanks for getting back to me and looking into this! I'll have to read into AOTs and test out the dev experience there before I can fully comment on that (just started using 3.0.0 and haven't had the opportunity to leverage them yet). However, one thing I would like to note is that for something like `transformOrigin`/`anchorOrigin`, I'd expect the values for `horizontal` and `vertical` to be polymorphic variants in line with the [material-ui-next docs](https://material-ui-next.com/api/popover/).

**Comment by jsiebern** [comment]
You're absolutely right, I'm quite busy at work right now but I'll try look into this as soon as possible!

**Comment by cknitt** [comment]
I have also run into troubles with these types, but with `Popover`, not with `Menu`.

When I try something like

```reason
<MaterialUi_Popover
  transformOrigin=(MaterialUi_Popover.makeTransformOrigin(~horizontal=30, ~vertical=30, ()))
  anchorOrigin=(MaterialUi_Popover.makeAnchorOrigin(~horizontal=30, ~vertical=30, ()))
  ...>
  ...
</MaterialUi_Popover>
```

I get an exception when opening the popover:

```
Uncaught TypeError: Cannot read property '0' of undefined
    at MaterialUi_Popover.bs.js:62
    at Object.map (js_option.js:62)
    at convertAnchorOrigin (MaterialUi_Popover.bs.js:61)
    at Object.make (MaterialUi_Popover.bs.js:193)
...
```

**Comment by jsiebern** [comment]
Thank you for the report! I hope to find some time this weekend for upgrading to the latest version and fixing those types!

**Comment by cknitt** [comment]
Thanks a lot! Yes, upgrading would be great, too, now that material-ui 1.0 is out. 😃

**Comment by cknitt** [comment]
Thanks for the update! 👍

**Comment by jsiebern** [comment]
Closing as it should be fixed in `v0.3.1`.
